module-2/backend/week_one.md

```diff
@@ -5,30 +5,82 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
-2. What is Sinatra?
+>get - client send a request to the server for info
+post - takes a `get` request and does the thing(make new thing) that it was asking for and then posts the new information to the page
+put - take a `get` request and does the thing(editing/updating) information on a page. and sends that info to the page
+delete - destroys the colum row.
+patch - is another way to `put` but not used as much as `put`
+
+2. What is Sinatra? 
+
+>DSL used in ruby to write web apps
+
 4. What is MVC?
+
+>*M*odel *V*iew *C*ontroller + routes
+
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+>So when future devs look/work on our code, they know what is going on.
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+>instance variables
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+
+```ruby
+  get '/horses' do
+    @name = 'Mr. Ed'
+    erb :index
+  end
+  ```
 9. What's the purpose of ERB?
+ > to be able to run ruby methods and return html.
+ 
 10. Why do I need a development AND test database?
+>Same reason you don't work on master branch for git. It keeps the "working" version pristine.
+
 11. What's responsive design?
+>changes based on screen size
+
 12. What is CRUD and why is it important?
+>*C*reate *R*ead *U*pdate *D*elete
+
 13. What does HTTP stand for? 
+>Hyper Text Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+><%= %> / <% %>, with the `=` it will show what is being evaluated, without will not.
+
 15. What's an ORM?
+>Object RElational Mapping
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+>SQL or Postgres
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+>get

```

Comment:
| **URL** | **HTTP Verb** |  **Action**|   **Explanation**|
 +|------------|-------------|------------|------------|
 +| /restaurants/         | GET       | index | See all restaurants.  
 +| /restaurants/new         | GET       | new | See form to enter a new restaurant.  
 +| /restaurants          | POST      | create | Submit form to create a new restaurant.  
 +| /restaurants/:id      | GET       | show | See one restaurant.      
 +| /restaurants/:id/edit | GET       | edit | See form to update a restaurant.      
 +| /restaurants/:id      | PATCH/PUT | update | Submit form to update a restaurant.   
 +| /restaurants/:id      | DELETE    | destroy | Delete a restaurant.